`kpt fn render` resource name missing in the result items in structured results

if you run `kpt fn render` on e2e/testdata/fn-render/kubeval-failure, you see the following output. Resource names are missing in the `Results` section. `Stderr` section contains the resource name correctly.

```
[FAIL] "gcr.io/kpt-fn/kubeval:unstable"
  Results:
    [ERROR] selector is required in object "apps/v1/Deployment/" in file "resources.yaml" in field "selector"
    [ERROR] template is required in object "apps/v1/Deployment/" in file "resources.yaml" in field "template"
    [ERROR] Invalid type. Expected: [integer,null], given: string in object "apps/v1/Deployment/" in file "resources.yaml" in field "spec.replicas"
    [ERROR] Validating arbitrary CRDs is not supported yet. You can skip them by setting ignore_missing_schemas or skip_kinds in the function config: ERR  - stdin: Failed initializing schema file:///tmp/master-standalone-strict/custom-custom-v1.json: open /tmp/master-standalone-strict/custom-custom-v1.json: no such file or directory  in object "custom.io/v1/Custom/" in file "resources.yaml"
  Stderr:
    "[ERROR] selector is required in object 'apps/v1/Deployment//nginx-deployment' in file resources.yaml in field selector"
    "[ERROR] template is required in object 'apps/v1/Deployment//nginx-deployment' in file resources.yaml in field template"
    "[ERROR] Invalid type. Expected: [integer,null], given: string in object 'apps/v1/Deployment//nginx-deployment' in file resources.yaml in field spec.replicas"
    "[ERROR] Validating arbitrary CRDs is not supported yet. You can skip them by setting ignore_missing_schemas or skip_kinds in the function config:"
    ...(3 line(s) truncated, use '--truncate-output=false' to disable)
  Exit code: 1
```

--- a/site/book/04-using-functions/03-function-results.md
+++ b/site/book/04-using-functions/03-function-results.md
@@ -59,7 +59,7 @@ Package "wordpress":
 [PASS] "gcr.io/kpt-fn/set-label:v0.1"
 [FAIL] "gcr.io/kpt-fn/kubeval:v0.1"
   Results:
-    [ERROR] Invalid type. Expected: integer, given: string in object "v1/Service/" in file "service.yaml" in field "spec.ports.0.port"
+    [ERROR] Invalid type. Expected: integer, given: string in object "v1/Service/wordpress" in file "service.yaml" in field "spec.ports.0.port"
   Exit code: 1
 
 For complete results, see /tmp/results.yaml
@@ -87,6 +87,7 @@ items:
         resourceRef:
           apiVersion: v1
           kind: Service
+          name: wordpress
         field:
           path: spec.ports.0.port
         file:

